pkg/util/node/node.go

```diff
@@ -164,3 +168,28 @@ func GetNodeIP(client clientset.Interface, name string) net.IP {
 	}
 	return nodeIP
 }
+
+// HasTaintByKey returns true if the node has the taint by the key passed in the argument.
+func HasTaintByKey(taintKey string, node v1.Node) bool {
```

Comment:
Check here:
https://github.com/kubernetes/kubernetes/blob/v1.24.0-alpha.3/pkg/util/taints/taints.go#L269

Reply:
This function checks for `taint key` as well as effect. I somehow assumed that we need to check only for `key`. 
But it makes sense that, here everything on taint will be triggered only if the taint has `out-of-service` key and effect as `NoExecute`

Reply:
Will make the changes